### ISSUE TITLE
Add a configuration file

### ISSUE THREAD

**Issue by Krzysztof-Cieslak** [LambdaFactory/Fornax#1]
Moved from: https://gitlab.com/Krzysztof-Cieslak/Fornax/issues/4

---

It's could be interesting to add config file to Fornax. I propose to use Toml over yaml because I got a lot of exception when using FsYaml (still trying to understand the reasons). Also, Toml seems better for this kind of usage.

Example of config file:

```toml
Exclude = [
  "node_modules/",
  "packages/"
]

[Style]
type = "sass" # Could by less, css
entry = "main.sass"
output = "bundle.css"
```

---------------

#### Exclude

Array of path to ignore during Fornax build. 

##### Why ?

Useful when people want to include CSS frameworks (via npm for example).

##### Current behavior

Read all the folders and build all the files fornax understand (less, sass, etc.)

##### Compatibility

If the option is not set, we keep the current behavior of Fornax.

---------------

#### Style

* Type: Type of the preprocessor to use
  - sass
  - less
  - css # Should probably not exist see "Compatibility" section
* Entry: Name of the entry file
* Output: Name of the output file

##### Why ?

In general, when using a preprocessor we have a bunch of files and only one entry point referencing the others. This setting allow this behavior and will improve the build time by fornax. Indeed, fornax will call the preprocessor only for the entry file and not all the files (who are useless).

##### Current behavior

Call the preprocessor depending on the file extension, and build each of the file found.
Take more time than needed, because it's process files that should be processed.

##### Compatibility

If the option is not set, we keep the current behavior of Fornax.

When people use `css` we could forgive to use this settings, as no processor is call. And let the current Fornax behavior.

---------------

The config file is **optional**, and if it does not exist then we keep the current behavior of fornax. Like so, we avoid breaking changes.

I can try to implement this features if you are ok with them.

**Comment by pblasucci** [comment]
I really like this idea. Especially the ability to exclude things. In fact, I was preparing a patch to address an issue in `0.2.0` whereby `_bin\Fornax.Core.dll` winds up in `_public`. However, I started thinking and realized a general exclusion mechanism is probably better. Seems you've already had the same thoughts @Krzysztof-Cieslak. Also, :100: for using TOML, as it feels like a better for for this sort of thing.

**Comment by Krzysztof-Cieslak** [comment]
`_bin` should definitely be just ignored without any additional config so feel free to send patch anyway. 

But yes, I plan to implement the config file one day ;-)

**Comment by pblasucci** [comment]
Coolio. Should I prepare an entry into `RELEASE_NOTES.md`, too? Or just the source change?

**Comment by Krzysztof-Cieslak** [comment]
Just the source change, I’ll handle updating release notes and release

**Comment by pblasucci** [comment]
Something like this: https://github.com/LambdaFactory/Fornax/pull/21 ?

**Comment by WalternativE** [comment]
I'd also say that TOML would most likely be the better solution (as the YAML spec is just very odd). Unfortunately though there just seems to be almost no real TOML support in the .NET space. I think I've found most TOML focused libraries on the web and they are either not maintained anymore, extremely inflexible or just terribly documented (often times all of the above).

YamlDotNet on the other hand is well designed, documented and appears to be stable. Would you all be ok with YAML as long as there is no better support in the ecosystem?

**Comment by Krzysztof-Cieslak** [comment]
Implemented in #40 - configuration file is `.fsx` script (for generator configuration) and/or any loader (for site global information)
